### PR TITLE
Modify pa_matchbatch.sh to work with private forks + add IGNORE_FOR and ONLY_ON

### PR DESCRIPTION
**Pull request by ekinanp**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -121,6 +121,17 @@ directory as the `pa_matchbatch.sh` script.
 `OVERRIDE_PATH` indicates where to find version overrides. See next section. It defaults to
 `/tmp/version_overrides.txt`.
 
+`IGNORE_REPOS` contains a (space-separated) list of repos that we do not want to process, i.e.
+run ticketmatch on. Each repo should be specified by their foss name (e.g. "puppet-agent" instead
+of "puppet-agent-private") since the script internally figures out whether to clone a foss or private
+fork. For example, IGNORE_REPOS="puppet-agent facter pxp-agent" will not run ticketmatch on puppet-agent,
+facter and pxp-agent. It defaults to empty.
+
+`ONLY_ON` contains a (space-separated) list of foss repos that we exclusively want to process, ignoring
```
Comment (ekinanp) [inline README.md]: This is useful when I want to run pa_matchbatch.sh on some of the repos. In my workflow when I am the release lead, I like to process things one at a time.

File: README.md [hunk 2]
```diff
@@ -121,6 +121,17 @@ directory as the `pa_matchbatch.sh` script.
 `OVERRIDE_PATH` indicates where to find version overrides. See next section. It defaults to
 `/tmp/version_overrides.txt`.
 
+`IGNORE_REPOS` contains a (space-separated) list of repos that we do not want to process, i.e.
```
Comment (ekinanp) [inline README.md]: Useful for when I've already checked a component, and do not want it cluttering the output.

Comment (ekinanp) [reply README.md]: Could maybe change this to IGNORE? IGNORE_REPOS might seem a bit redundant. It would also be more consistent with ONLY_ON.

Comment (speedofdark) [reply README.md]: If we're trying to stay in the spirit of ONLY_ON, then I would name this IGNORE_FOR

File: pa_matchbatch.sh [hunk 3]
```diff
@@ -25,6 +25,31 @@ popd() {
...
+# parses the repo name from a given git URL
+parseRepoName() {
+	basename ${1} .git
+}
+
+# checks if an array contains the given element
+containsElement() {
+	local array="$1"
+	local element="$2"
+
```
Comment (speedofdark) [inline pa_matchbatch.sh]: For consistency, ${1} and ${2}, please?

Comment (ekinanp) [reply pa_matchbatch.sh]: Updated

File: README.md (hunk 2 above)
Comment (ekinanp) [reply README.md]: That is much better. Updated.

**Comment by ekinanp** [comment]
Updated the PR per your comments @speedofdark

**Comment by speedofdark** [comment]
Unless you feel it's not warranted, I'm ok with merging this.

**Review by caseywilliams — APPROVED** [review]
Seems fine

**Pull request merged**